witness/src/Dscp/Witness/CLI.hs

```diff
@@ -101,17 +114,26 @@ committeeParamsParser =
     commSecretParser = option committeeSecretReadM
         (long "comm-sec" <>
          metavar "BYTESTRING" <>
-         help "Committee secret key. Common key for the core nodes \
-              \which serves as root key in generation of participants' secrets.")
+         help "Committee secret key. Common key for the core nodes which serves\
+              \ as root key in generation of participants' secrets. To use an \
+              \open committee, provide the `--comm-open` flag instead.")
 
 -- | CLI parser for witness key params.
--- Note [DSCP-347]: there is a bug -- one cannot redefine committee secret keys
--- with basic key params in CLI. Such usecase is not very likely, but
--- this is still counterintuitive.
-witnessKeyParamsParser :: ModParser WitnessKeyParams
-witnessKeyParamsParser =
-    modifying (Committee <$> committeeParamsParser) <|>
-    over _Basic <$> baseKeyParamsParser "witness"
+witnessKeysParamsParser :: OptModParser WitnessKeysParams
+witnessKeysParamsParser = #params .:+
+    (#paramsType %:: (basicP <|> committeeP) <*<
+     #basic      .:- baseKeyParamsParser "witness" <*<
+     #committee  .:- committeeKeyParamsParser
+    )
+  where
+    basicP = flag' (const "basic") $
```

Comment:
Why do you return `const` functions, if you can just return values and use `.::` instead of `%::`?

Reply:
I honestly don't recall my reasoning behind this, I think I was trying something else and left it like this by mistake.
You are right, I'll fix it.